pygac_fdr/tests/test_end2end.py

```diff
@@ -0,0 +1,470 @@
+from pygac_fdr.utils import logging_on, LOGGER_NAME
+
+
+LOG = logging.getLogger(LOGGER_NAME)
+
+
+def call_subproc(cmd, cwd='.'):
+    LOG.info('Calling subprocess {}'.format(cmd))
+    ret = subprocess.call(cmd, cwd=cwd)
+    if ret:
+        raise RuntimeError('Subprocess {} failed with return code {}'.format(cmd, ret))
+
+
+class EndToEndTestBase(unittest.TestCase):
+    def _assert_time_attrs_close(self, attrs_a, attrs_b):
+        time_attrs = ['start_time', 'end_time']
+        for attr in time_attrs:
+            with self.subTest(attribute=attr):
+                time_a = isoparse(attrs_a.pop(attr))
+                time_b = isoparse(attrs_b.pop(attr))
+                self.assertLess(abs((time_b - time_a).total_seconds()), 2)
+
+    def _assert_numerical_attrs_close(self, attrs_a, attrs_b):
+        numerical_attrs = [
+            'geospatial_lon_min',
+            'geospatial_lon_max',
+            'geospatial_lat_min',
+            'geospatial_lat_max',
+        ]
+        for attr in numerical_attrs:
+            with self.subTest(attribute=attr):
+                val_a = attrs_a.pop(attr)
+                val_b = attrs_b.pop(attr)
+                np.testing.assert_allclose(val_a, val_b)
+
+    def assert_attrs_close(self, a, b):
+        attrs_a = a.attrs.copy()
+        attrs_b = b.attrs.copy()
+        self._assert_time_attrs_close(attrs_a, attrs_b)
+        self._assert_numerical_attrs_close(attrs_a, attrs_b)
+        assert dict_equiv(attrs_a, attrs_b), diff_attrs_repr(attrs_a, attrs_b, 'identical')
+
+    @classmethod
+    def setUpClass(cls):
+        logging_on(logging.DEBUG, for_all=True)
+
+        # Read config file
+        with open(os.path.join(os.path.dirname(__file__), 'test_end2end.yaml')) as fh:
+            config = yaml.safe_load(fh)
+        cls.test_data_dir = config['test_data_dir']
+        cls.cleanup = config['cleanup']
+        cls.resume = config['resume']
+        cls.fast = config['fast']
+        cls.rtol = float(config['rtol'])
+        cls.atol = float(config['atol'])
+        cls.plot = config['plot']
+
+        # Set class attributes
+        cls.cfg_file = os.path.join(os.path.dirname(__file__), '../../etc/pygac-fdr.yaml')
+        cls.tle_dir = os.path.join(cls.test_data_dir, 'tle')
+        cls.input_dir = os.path.join(cls.test_data_dir, 'input')
+        cls.output_dir = None  # to be set by subclasses
+        cls.output_dir_ref = os.path.join(cls.test_data_dir, 'output_ref')
+
+    @classmethod
+    def _unzip_gac_files(cls, filenames_gz, output_dir):
+        filenames = []
+        for filename_gz in filenames_gz:
+            basename_gz = os.path.basename(filename_gz)
+            basename = os.path.splitext(basename_gz)[0]
+            filename = os.path.join(output_dir, os.path.basename(basename))
+            LOG.info('Decompressing {}'.format(filename_gz))
+            with gzip.open(filename_gz, 'rb') as f_in:
+                with open(filename, 'wb') as f_out:
```

Comment:
multiple with statements can be combined with `,`. Example:
```python
with open(path_1) as file_1, open(path_2) as file2:
    do_something(file_1, file_2)
```
In case of exceeding the maximum line length, consider https://www.python.org/dev/peps/pep-0008/#maximum-line-length